Sources/NIO/Bootstrap.swift

```diff
@@ -966,4 +1133,134 @@ public final class NIOPipeBootstrap {
             }
         }
     }
+    
+    /// A channel option which can be applied to pipe bootstrap using shorthand notation.
+    /// - See: NIOPipeBootstrap.channelOptions(_ options: [Option])
+    public struct Option {
+        private let data: ShorthandChannelOption
+        
+        private init(_ data: ShorthandChannelOption) {
+            self.data = data
+        }
+        
+        /// Apply the contained option to the supplied NIOPipeBootstrap
+        /// - Parameter to: bootstrap to apply this option to.
+        @usableFromInline
+        func applyOption(to bootstrap: NIOPipeBootstrap) {
+            data.applyOption(to: bootstrap)
+        }
+        
+        fileprivate enum ShorthandChannelOption {
+            case disableAutoRead
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to bootstrap: NIOPipeBootstrap) {
+                switch self {
+                case .disableAutoRead:
+                    _ = bootstrap.channelOption(ChannelOptions.autoRead, value: false)
+                case .allowRemoteHalfClosure(let value):
+                    _ = bootstrap.channelOption(ChannelOptions.allowRemoteHalfClosure, value: value)
+                }
+            }
+        }
+    }
+}
+
+// Hashable for the convenience of users.
+extension NIOPipeBootstrap.Option: Hashable {}
+extension NIOPipeBootstrap.Option.ShorthandChannelOption: Hashable {}
+
+/// Approved shorthand datagram channel options.
+extension NIOPipeBootstrap.Option {
+    /// Option to disable autoRead
+    /// - See: ChannelOptions.autoRead
+    public static let disableAutoRead = NIOPipeBootstrap.Option(.disableAutoRead)
+    
+    /// - See: `AllowRemoteHalfClosureOption`.
+    public static let allowRemoteHalfClosure =
+        NIOPipeBootstrap.Option(.allowRemoteHalfClosure(true))
+    
+    /// - See: `AllowRemoteHalfClosureOption`.
+    public static func allowRemoteHalfClosure(_ value: Bool) ->
+        NIOPipeBootstrap.Option {
+        return NIOPipeBootstrap.Option(.allowRemoteHalfClosure(value))
+    }
+}
+
+
+
+// ------------------------
+
+public protocol NIOTCPOptionAppliable {
```

Comment:
Can we have a doc comment explaining what this protocol is for?

Reply:
Yes, and rethought a little more like a closure.